service/model_manager/model_list.py

```diff
@@ -0,0 +1,47 @@
+import os
+import folder_paths
+import server
+from aiohttp import web
+import hashlib
+
+comfy_path = os.path.dirname(folder_paths.__file__)
+
+model_root_dir = os.path.join(comfy_path, 'models')
+
+
+# save file hash dict
+file_hash_dict = {}
+
+@server.PromptServer.instance.routes.get("/model_manager/get_model_list")
```

Comment:
you should add a subpath, when user is on checkpoint tab, only read checkpoint subfolder, because calculating hash is reading the whole file, some checkpoint files are 5GB, it is very time consuming, you should not read the entire models/ dir all at once